### DISCUSSION TITLE
How to get the feature flags of useReactTable instance in columnDefs?

### DISCUSSION THREAD

**Discussion by mikhaelesa** [TanStack/table#5435] [Q&A]
So I have this useReactTable instance with some feature flags set

```
const table = useReactTable({
    columns: MY_COLUMNS,
    data: MY_DATA,
    state: {
      rowSelection,
    },
    enableRowSelection: true,
    enableMultiRowSelection: false, // I don't want multi select hence the flag set to false
    onRowSelectionChange: setRowSelection,
    getCoreRowModel: getCoreRowModel(),
  });
```

And this is my columndef

```
[
  {
    id: "selection",
    enableResizing: false,
    size: 24,
    header: ({ table }) => (
        <Checkbox
          {...{
            checked: table.getIsAllRowsSelected(),
            indeterminate: table.getIsSomeRowsSelected(),
            onChange: table.getToggleAllRowsSelectedHandler(),
          }}
        />
    ),
    cell: ({ row }) => (
        <Checkbox
          {...{
            checked: row.getIsSelected(),
            disabled: !row.getCanSelect(),
            onChange: row.getToggleSelectedHandler(),
          }}
        />
    ),
  },
  // ...other columns
];
```

Since I use this column in multiple places, I really want to use the same column def instead of recreating it again. The problem is that some places don't need to have multiple row selection. Since the column def allows this feature by default (See the onChange), it doesn't allow me to do that.

If there's only some way to get the `enableMultiRowSelection` flag through the column def, then I can just do a simple ternary condition based on that.

Is there any solution to this? Thanks in advance!

**Comment by Xxs-7** [comment]
set `enableMultiRowSelection` to be a function may solve your problem，return false or true by condition in the function.
`enableMultiRowSelection` is union type: `enableMultiRowSelection?: boolean | ((row: Row<TData>) => boolean)`
getCanMultiSelect will call the given function inside.
``` ts
 row.getCanMultiSelect = () => {
      if (typeof table.options.enableMultiRowSelection === 'function') {
        return table.options.enableMultiRowSelection(row)
      }

      return table.options.enableMultiRowSelection ?? true
    }
```
